nxdl/nxdl_to_hdf5.py

```diff
@@ -782,24 +799,25 @@ def process_symbols(soup, sym_args_dct={}):
 def get_extending_class(fname, cls_nm, sym_args_dct={}, dct={}, docs=[], report_symbols_only=False):
     fparts = fname.split('\\')
     eclass_file = fname.replace(fparts[-1], '%s.nxdl.xml' % cls_nm)
-    if (not os.path.exists(eclass_file)):
+
+    if not pathlib.Path(eclass_file).exists():
         print('get_extending_class: XML file [%s] does not exist' % eclass_file)
         return({}, {}, {})
     print('extending with [%s]' % cls_nm)
     dct, syms, docs = get_xml_paths(eclass_file, sym_args_dct=sym_args_dct, dct=dct, docs=docs, report_symbols_only=report_symbols_only, allow_extend=True)
 
     return (dct, syms, docs)
 
-def get_xml_root(fname):
+def get_xml_root(fpath):
     '''
     takes the path to teh nxdl.xml file and returns a dict of element category lists of the entire structure
     '''
 
-    if(not fname.exists()):
-        print('XML file [%s] does not exist' % str(fname.absolute()))
+    if not fpath.exists():
+        print('XML file [%s] does not exist' % str(fpath))
         return(None)
 
-    infile = open(fname, "r")
+    infile = open(str(fpath), "r")

```

Comment:
switch these three lines to a _context_ (it closes the file automatically, even if there is an exception):

```python
    with open(str(fpath), "r") as infile:
        contents = infile.read()
```

Reply:
👍